variables.tf

```diff
@@ -0,0 +1,238 @@
+  type        = string
+}
+
+variable "domains" {
+  description = "A set of Domain names to serve as entry points for your Service."
+  type = list(object({
+    name    = string
+    comment = optional(string, "")
+  }))
+}
+
+variable "activate" {
+  description = "Conditionally prevents the Service from being activated. The apply step will continue to create a new draft version but will not activate it if this is set to false."
+  type        = bool
+  default     = true
+}
+
+variable "default_ttl" {
+  description = "The default Time-to-live (TTL) for requests."
+  type        = number
+  default     = 3600
+}
+
+variable "http3" {
+  description = "Enables support for the HTTP/3 (QUIC) protocol."
+  type        = bool
+  default     = true
+}
+
+variable "force_tls_enable_hsts" {
+  description = "Force TLS and HTTP Strict Transport Security (HSTS). Creates necessary objects in a similar fashion to the GUI flag."
+  type        = bool
+  default     = true
+}
+
+variable "hsts_duration" {
+  description = "How long the client should remember that the domain uses HSTS"
+  type        = number
+  default     = 31557600
+}
+
+variable "stale_if_error" {
+  description = "Enables serving a stale object if there is an error."
+  type        = bool
+  default     = true
+}
+
+# Backends
+variable "backends" {
+  description = "All backends that will be used for this service."
+  type = list(object({
+    name    = string
+    address = string
+
+    auto_loadbalance  = optional(bool, true)
+    healthcheck       = optional(string, "")
+    max_conn          = optional(number, 200)
+    max_tls_version   = optional(string, "")
+    min_tls_version   = optional(string, "1.2")
+    port              = optional(number, 443)
+    shield            = optional(string, "")
+    ssl_ca_cert       = optional(string, "")
+    ssl_cert_hostname = optional(string, "")
+    ssl_sni_hostname  = optional(string, "")
+    override_host     = optional(string, "")
+    use_ssl           = optional(bool, true)
+  }))
+  default = []
+}
+
+variable "backend_ssl_ca_cert" {
+  description = "CA certificate attached to origin. Used to pull value from Terraform Cloud."
+  type        = string
+  sensitive   = true
+  default     = ""
+}
+
+# Healthcheck endpoints.
+variable "healthchecks" {
+  description = "Endpoints to periodically test for backend health."
+  type = list(object({
+    name = string
+    host = string
+    path = string
+
+    check_interval    = optional(number, 60000)
+    expected_response = optional(number, 200)
+    headers           = optional(list(string), [])
+    http_version      = optional(string, "1.1")
+    initial           = optional(number, 1)
+    method            = optional(string, "GET")
+    threshold         = optional(number, 1)
+    timeout           = optional(number, 5000)
+    window            = optional(number, 2)
+  }))
+  default = []
+}
+
+# Custom headers
+variable "headers" {
```

Comment:
What is this used for?

Reply:
This is equivalent to the [Headers configuration](https://docs.fastly.com/en/guides/adding-or-modifying-headers-on-http-requests-and-responses) that you can add to services. There are a couple services that have custom headers specified (for example, `http.Fastly-Purge-Requires-Auth` for the `files-*` services)